test/pool.test.ts

```diff
@@ -45,30 +80,103 @@ abstract class PoolUser {
 
   async setReceiver(receiver: this, weight: number): Promise<void> {
     const receivers = await this.getAllReceivers();
+
     await submit(this.pool.setReceiver(receiver.addr, weight), "setReceiver");
-    if (weight == 0) {
-      receivers.delete(receiver.addr);
-    } else {
-      receivers.set(receiver.addr, weight);
-    }
-    await this.expectReceivers(receivers);
-  }
 
-  async getAllReceivers(): Promise<Map<string, number>> {
-    const receivers = await this.pool.getAllReceivers();
-    return new Map(receivers.map(({receiver, weight}) => [receiver, weight]));
+    updateReceiversList(
+      receivers,
+      receiver,
+      (receiverWeight) => (receiverWeight.receiverWeight = weight)
+    );
+    await this.expectReceivers(receivers);
   }
 
   async expectSetReceiverReverts(
     receiver: this,
     weight: number,
     expectedCause: string
   ): Promise<void> {
+    const receivers = await this.getAllReceivers();
     await submitFailing(
       this.pool.setReceiver(receiver.addr, weight),
       "setReceiver",
       expectedCause
     );
+    await this.expectReceivers(receivers);
+  }
+
+  async setProxy(receiver: this, weight: number): Promise<void> {
+    const receivers = await this.getAllReceivers();
+    await submit(this.pool.setProxy(receiver.addr, weight), "setProxy");
+    updateReceiversList(
+      receivers,
+      receiver,
+      (receiverWeight) => (receiverWeight.proxyWeight = weight)
+    );
+    await this.expectReceivers(receivers);
+  }
+
+  async expectSetProxyReverts(
+    receiver: this,
+    weight: number,
+    expectedCause: string
+  ): Promise<void> {
+    const receivers = await this.getAllReceivers();
+    await submitFailing(
+      this.pool.setProxy(receiver.addr, weight),
+      "setProxy",
+      expectedCause
+    );
+    await this.expectReceivers(receivers);
+  }
+
+  async getAllReceivers(): Promise<ProxyReceiverWeights> {
+    const receivers = await this.pool.getAllReceivers();
+    return new Map(
+      receivers.map(({receiver, receiverWeight, proxyWeight}) => [
+        receiver,
+        new ProxyReceiverWeight(receiverWeight, proxyWeight),
+      ])
+    );
+  }
+
+  async setProxyWeights(weights: [this, number][]): Promise<void> {
```

Comment:
Similar to what I said above: We only need the address from each weight. To avoid confusion I’d prefer
```suggestion
  async setProxyWeights(weights: [string, number][]): Promise<void> {
```

Reply:
I'd very much prefer not to do that. This will require extracting the address for every proxy receiver in every test, which is a ton of unnecessary noise in the test bodies.

Reply:
Fair enough